### PR TITLE
Add expat and libxml2 to list of allowed version ranges

### PR DESCRIPTION
**Pull request by RubenRBS**
Document bounds for libxml2 and expat

/cc @mayeut who's been pushing for this lately, thanks!

### CODE REVIEW

**Review by ericLemanissier — CHANGES_REQUESTED** [review]
is libpng not allowed any more ?

File: docs/adding_packages/dependencies.md [hunk 1]
```diff
@@ -178,14 +178,15 @@ for consumer, we do impose some limits on Conan features to provide a smoother f
...
+Currently, these are (except if the recipe needs a newer lower version for specific reasons):
 
 * OpenSSL: `[>=1.1 <4]` for libraries known to be compatible with OpenSSL 1.x and 3.x
 * CMake: `[>3.XX <4]`, where `3.XX` is the minimum version of CMake required by the relevant build scripts. Note that CCI recipes assume 3.15 is installed in the system, so add this
 version range only when a requirement for a newer version is needed.
-* Libcurl: `[>=X.YY <9]`, where `X.YY` is the minimum version of Libcurl required, starting from `7.78`
-* Zlib: `[>=1.2.11 <2]` expect if the recipe needs a newer lower version for specific reasons
-* Libpng: `[>=1.6 <2]` expect if the recipe needs a newer lower version for specific reasons
+* Libcurl: `[>=7.78 <9]`
+* Zlib: `[>=1.2.11 <2]`
```
Comment (RubenRBS) [inline docs/adding_packages/dependencies.md:187]:
```suggestion
* Zlib: `[>=1.2.11 <2]`
* Libpng: `[>=1.6 <2]`
```

**Review by ericLemanissier — APPROVED** [review]

File: docs/adding_packages/dependencies.md [hunk 2]
```diff
@@ -178,14 +178,16 @@ for consumer, we do impose some limits on Conan features to provide a smoother f
...
+* Libpng: `[>=1.6 <2]`
+* Expat: `[>=2.6.2 <3]`
+* Libxml2: `[>=2.12.5 <3]`
```
Comment (SpaceIm) [inline docs/adding_packages/dependencies.md:190]: It there a rational for such high lower bound? I know that some recipes are not compatible with libxml2 2.12.x due to removal of few functions.

Comment (RubenRBS) [reply docs/adding_packages/dependencies.md:190]: Hi, this is just the version that had no cve at the time of writing

Note that when necessary, we can nudge the verison ranges to fit when a library does not support newer versions, I'll add a note, thanks for the heads up :)

Comment (RubenRBS) [reply docs/adding_packages/dependencies.md:190]: Pinging @SpaceIm in case you don't get answer notifications!

**Pull request merged**